### ISSUE TITLE
Extension of type 'Task' does not exist. (v2.0.5)

### ISSUE THREAD

**Issue by RoccoC** [PeredurOmega/GradleNodePlugin#15]
I'm seeing an error in my project when upgrading to version `2.0.5` of the plugin:

`Extension of type 'Task' does not exist. Currently registered extension types: [ExtraPropertiesExtension]`.

This can be reproduced in the `sample` project by making the following changes to `sample/build.gradle.kts`:

```diff
diff --git a/sample/build.gradle.kts b/sample/build.gradle.kts
index 7d03d64..7e381e8 100644
--- a/sample/build.gradle.kts
+++ b/sample/build.gradle.kts
@@ -6,6 +6,15 @@ repositories {
     nodeRepository()
 }
 
+tasks.register("clean") {
+    logger.lifecycle("clean task is running")
+}
+
 node {
     packageManager.set(PackageManager.PNPM)
+    cleanTaskName.set("nodeClean")
+}
+
+tasks.named<NodeCleanTask>("nodeClean") {
+    logger.lifecycle("nodeClean task is running")
 }
\ No newline at end of file
```
This looks to be related to a [recent change](https://github.com/PeredurOmega/GradleNodePlugin/commit/7d93df557fc1821e56fd497c19e5239cad199006) and stems from the call to `configure`, [here](https://github.com/PeredurOmega/GradleNodePlugin/commit/7d93df557fc1821e56fd497c19e5239cad199006#diff-dcf97b4038bedacdbd2308c0ac880e221888b3ef0b5bc3bcccdad455deb01f36R120), though I am not experienced enough with gradle to understand the root cause. 😢 

The below fixes things, but again I am not sure why `configure` isn`t working here: 🤷 

```diff
diff --git a/src/main/kotlin/NodePlugin.kt b/src/main/kotlin/NodePlugin.kt
index 0a9c205..2b13329 100644
--- a/src/main/kotlin/NodePlugin.kt
+++ b/src/main/kotlin/NodePlugin.kt
@@ -3,7 +3,6 @@ import com.google.gson.JsonObject
 import nu.studer.gradle.credentials.CredentialsPlugin
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.Task
 import org.gradle.api.plugins.BasePlugin
 import org.gradle.api.provider.Provider
 import org.gradle.kotlin.dsl.*
@@ -115,9 +114,7 @@ class NodePlugin : Plugin<Project> {
             // package.json
             if (extension.cleanTaskName.get() != BasePlugin.CLEAN_TASK_NAME) {
                 val baseCleanTask = project.tasks.findByName(BasePlugin.CLEAN_TASK_NAME)
-                baseCleanTask?.configure<Task> {
-                    finalizedBy(extension.cleanTaskName.get())
-                }
+                baseCleanTask?.finalizedBy(extension.cleanTaskName.get())
             }
         }
```

Happy to raise a PR if this change seems reasonable! Thank you! 🙏

**Comment by PeredurOmega** [comment]
Hey @RoccoC !

Thanks a lot for raising this issue and providing a detail way to reproduce it and proposing a fix.

Referring to the [Gradle Doc](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview), [findByName](https://docs.gradle.org/current/javadoc/org/gradle/api/NamedDomainObjectCollection.html#findByName-java.lang.String-) is part of the Eager API, which should be avoided. So I replaced it with [named](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-) and caught the associated error in case of a missing task, as this task won't impact the plugin in any case if it doesn't exist in the project. 

This issue is now fixed with [2.0.6](https://github.com/PeredurOmega/GradleNodePlugin/releases/tag/2.0.6).

**Comment by RoccoC** [comment]
Excellent, thanks so much!
